File: package/backend/movie-service/src/services/manager.rs
use crate::services::movie_service::MovieSrv;
use crate::store::store::Store;

pub struct Manager {
    pub user_service: MovieSrv
}

impl Manager {
    pub fn new(store: Store) -> Manager {
        Manager {
            user_service: MovieSrv::new(store)
        }
    }
}

File: package/backend/user-service/pkg/services/manager.go
package services

import (
	"github.com/user-service/pkg/store"
)

// Manager is just a collection of all services we have in the project
type Manager struct {
	User UserService
}

// NewManager creates new service manager
func NewManager(store *store.Store) *Manager {
	return &Manager{
		User: NewUserService(store.User),
	}
}
